fact-ebpf/src/bpf/checks.c

```diff
@@ -0,0 +1,15 @@
+// clang-format off
+#include "vmlinux.h"
+
+#include "bound_path.h"
+
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+// clang-format on
+
+SEC("lsm/path_unlink")
+int BPF_PROG(check_path_unlink_supports_bpf_d_path, struct path* dir, struct dentry* dentry) {
```

Comment:
I think there is another option. From what I understand we know [the exact commit](https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=b13cddf633562b9b2c34fd63471d377019704ebe) (which on bpf-next resolves to v6.7-rc3), where an LSM hook on path_unlink became capable of calling bpf_d_path. How about just doing a condition on the kernel version?